### PR TITLE
HPCC-18284 Problem with ECLCC after C++11 support merged.

### PR DESCRIPTION
**Pull request by AttilaVamos**
Query the current g++ path from the Linux environmnet

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:

Tested manually on single gcc version
   - Ubuntu 16.04: gcc (Ubuntu 5.4.0-6ubuntu1~16.04.5) 5.4.0 20160609
   - CentOS 7.x: gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-4))

and multiple gcc versions (CentOS 6.x)
   - gcc (GCC) 4.4.7 20120313 (Red Hat 4.4.7-18)
   - gcc (GCC) 4.8.2 20140120 (Red Hat 4.8.2-15)
environments

### CODE REVIEW

**Comment by AttilaVamos** [comment]
@richardkchapman please review.

File: system/jlib/jcomp.cpp [hunk 1]
```diff
@@ -163,17 +166,53 @@ static void doSetCompilerPath(const char * path, const char * includes, const ch
...
+        FILE* fp = popen(typeCmd.str(), "r");
+        if (fp != nullptr)
+        {
+            fgets(result, resultLength - 1, fp);
+            pclose(fp);
+        }
+#ifdef _DEBUG
+        PrintLog("'%s' command result is: '%s'", typeCmd.str(), result);
+#endif
+        const char* pathStart = strstr(result, PATHSEPSTR);
```
Comment (richardkchapman) [inline system/jlib/jcomp.cpp]: Use strchr(PATHSEPCHR) instead, will simplify some of this!

**Comment by richardkchapman** [comment]
Unless I'm missing something, this change is using "type g++" to locate the g++ compiler's full path, then executes it there. Wouldn't you get the same effect by simply removing the hard-coded /usr/bin on the compiler path and executing g++ ?

**Comment by richardkchapman** [comment]
If for some reason we can't just execute g++ (without a path) as suggested above, using command -v rather than type will probably give a result that is easier to parse.

@AttilaVamos Back to you

**Comment by AttilaVamos** [comment]
@richardkchapman updated, please re-review.

**Comment by richardkchapman** [comment]
I remain unconvinced that this change is a good idea. Setting CL_PATH=/opt/rh/devtoolset-2/root/usr on your centos6 box would have avoided the need for it...

**Comment by richardkchapman** [comment]
(the code in jcomp is horrible, and could do with a thorough overhaul and simplification at some point - but this proposed patch only makes that worse)

**Comment by AttilaVamos** [comment]
I can handle the multiple compiler sets with setting CL_PATH=/opt/rh/devtoolset-2/root/usr therefore this changes not necessary and I close  this PR..

**Pull request closed without merging**